tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc

```diff
@@ -1797,10 +1797,17 @@ Status TileTransposer::TransposeNode(TransposeContext* context,
 Status UnaryGradTransposer::TransposeNode(TransposeContext* context,
                                           utils::MutableNodeView* node) {
   DCHECK(IsUnaryGrad(*node->node()));
-  if (!ShouldProcess(*context, *node) || !IsFanoutPortRankN(*node, 0, 4) ||
+  const auto* output_shape_attr = node->GetAttr(kAttrOutputShape);
+  const auto& shape = output_shape_attr->list().shape(0);
+  const int rank = shape.dim_size();
```

Comment:
Can this be replaced with using `GetFanoutPortRank` (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc#L736-L739) along with an explicit rank size check? Currently this does not restrict the transposer to only allow for rank 4 and rank 5 tensors.